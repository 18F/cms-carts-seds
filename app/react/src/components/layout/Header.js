import React, { Component } from "react";

class Header extends Component {
  constructor() {
    super();

    this.toggleUserNav = this.toggleUserNav.bind(this);
  }

  toggleUserNav(e) {
    e.preventDefault();
    document.getElementById("menu-block").classList.toggle("open");
    document.getElementById("nav-user").classList.toggle("open");

    // Close menu when leaving focus
    let root = document.getElementById("root");
    root.addEventListener(
      "click",
      () => {
        document.getElementById("menu-block").classList.remove("open");
        document.getElementById("nav-user").classList.remove("open");
      },
      false
    );
  }

  render() {
    return (
      <div className="header">
        <div className="ds-l-container">
          <div className="ds-l-row header-row">
            <div className="site-title ds-l-col--6 ds-u-padding--2">
              <a href="/">Carts</a>
            </div>
            <div className="user-details ds-l-col--6 ds-u-padding--2">
              <div className="ds-l-row">
                <div className="save-status ds-l-col--6">Autosaved</div>
                <div className="nav-user ds-l-col--6" id="nav-user">
                  <ul className="user-email-button">
                    <li>
                      <a href="#menu" onClick={this.toggleUserNav.bind(this)}>
                        karen.dalton@state.gov
                      </a>
                    </li>
                  </ul>
                  <ul className="menu-block" id="menu-block">
                    <li className="manage-account">
                      <a href="#">Manage account</a>
                    </li>
                    <li className="logout">
                      <a href="#">Log out</a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Header;
