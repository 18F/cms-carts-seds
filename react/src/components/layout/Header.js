import React, { Component } from "react";

class Header extends Component {
  render() {
    return (
      <div className="header">
        <div className="ds-l-container">
          <div class="ds-l-row">
            <div className="site-title ds-l-col--6 ds-u-padding--2">
              <a href="/">Carts</a>
            </div>
            <div className="user-details ds-l-col--6 ds-u-padding--2">
              <div className="save-status">Autosaved</div>
              <div className="nav-user">
                <ul>
                  <li>
                    karen.dalton@state.gov
                    <ul>
                      <li className="manage-account">
                        <a href="#">Manage account</a>
                      </li>
                      <li className="logout">
                        <a href="#">Log out</a>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Header;
