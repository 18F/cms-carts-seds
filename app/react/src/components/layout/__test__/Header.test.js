import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";

import configureMockStore from "redux-mock-store";

import Header from "../Header";

const mockStore = configureMockStore();

describe("Header Component, enzyme testing", () => {
  const store = mockStore({
    stateUser: {
      currentUser: {
        role: "admin",
        state: { id: "NY", name: "New York" },
        username: "karen.dalton@state.gov",
      },
    },
  });

  const wrapper = mount(
    <Provider store={store}>
      <Header />
    </Provider>
  );
  it("renders", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("renders with header classname", () => {
    expect(wrapper.exists(".header")).toBe(true);
  });
});

describe("Header Component, snapshot testing", () => {
  const store = mockStore({
    stateUser: {
      currentUser: {
        role: "admin",
        state: { id: "NY", name: "New York" },
        username: "karen.dalton@state.gov",
      },
    },
  });

  const snapShot = renderer.create(
    <Provider store={store}>
      <Header />
    </Provider>
  );

  it("should render with given state from Redux store", () => {
    expect(snapShot.toJSON()).toMatchSnapshot();
  });
});
