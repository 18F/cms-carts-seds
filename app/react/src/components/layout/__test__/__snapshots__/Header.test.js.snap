// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component, enzyme testing should render with given state from Redux store 1`] = `
<div
  className="header"
>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row header-row"
    >
      <div
        className="site-title ds-l-col--6 ds-u-padding--2"
      >
        <a
          href="/"
        >
          Carts
        </a>
      </div>
      <div
        className="user-details ds-l-col--6 ds-u-padding--2"
      >
        <div
          className="ds-l-row"
        >
          <div
            className="save-status ds-l-col--6"
          >
            Autosaved
          </div>
          <div
            className="nav-user ds-l-col--6"
            id="nav-user"
          >
            <ul
              className="user-email-button"
            >
              <li>
                <a
                  className="nav--dropdown__trigger"
                  href="#menu"
                  onClick={[Function]}
                >
                  karen.dalton@state.gov
                </a>
              </li>
            </ul>
            <ul
              className="menu-block"
              id="menu-block"
            >
              <li
                className="manage-account"
              >
                <a
                  href="javascript:void(0)"
                >
                  Manage account
                </a>
              </li>
              <li
                className="logout"
              >
                <a
                  href="javascript:void(0)"
                >
                  Log out
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Component, snapshot testing should render with given state from Redux store 1`] = `
<div
  className="header"
>
  <div
    className="ds-l-container"
  >
    <div
      className="ds-l-row header-row"
    >
      <div
        className="site-title ds-l-col--6 ds-u-padding--2"
      >
        <a
          href="/"
        >
          Carts
        </a>
      </div>
      <div
        className="user-details ds-l-col--6 ds-u-padding--2"
      >
        <div
          className="ds-l-row"
        >
          <div
            className="save-status ds-l-col--6"
          >
            Autosaved
          </div>
          <div
            className="nav-user ds-l-col--6"
            id="nav-user"
          >
            <ul
              className="user-email-button"
            >
              <li>
                <a
                  className="nav--dropdown__trigger"
                  href="#menu"
                  onClick={[Function]}
                >
                  karen.dalton@state.gov
                </a>
              </li>
            </ul>
            <ul
              className="menu-block"
              id="menu-block"
            >
              <li
                className="manage-account"
              >
                <a
                  href="javascript:void(0)"
                >
                  Manage account
                </a>
              </li>
              <li
                className="logout"
              >
                <a
                  href="javascript:void(0)"
                >
                  Log out
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
